zeebe/broker-client/src/main/java/io/camunda/zeebe/broker/client/impl/BrokerRequestManager.java

```diff
@@ -287,6 +300,20 @@ static RequestResult failed(final ErrorCode code) {
     }
   }
 
+  private final record ResponseValidationWithPredicate<T>(
+      BrokerRequest<T> request, Predicate<T> predicate) implements Predicate<DirectBuffer> {
+
+    @Override
+    public boolean test(final DirectBuffer responseBytes) {
+      if (BrokerRequestManager.responseValidation(responseBytes)) {
+        final var response = request.getResponse(responseBytes);
```

Comment:
Unfortunately this will deserialize twice, I'll try to make it more efficient in a followup PR